spec/services/forest_liana/resources_getter_spec.rb

```diff
@@ -54,11 +68,80 @@ def init_scopes
       init_scopes
     end
 
-    after(:each) do
-      User.destroy_all
-      Island.destroy_all
-      Location.destroy_all
-      Tree.destroy_all
+    describe 'records eager loading' do
+      let(:resource) { Product }
+      let(:fields) { { resource.name => 'id,name,manufacturer', 'manufacturer' => 'name' } }
+
+      before do
+        orange_manufacturer = Manufacturer.create!(name: 'Orange')
+        pear_manufacturer = Manufacturer.create!(name: 'Pear')
+
+        baby_driver = Driver.create!(firstname: 'Baby driver')
+        taxi_driver = Driver.create!(firstname: 'Taxi driver')
+
+        Product.create!(name: 'Valencia', uri: 'https://valencia.com', manufacturer: orange_manufacturer, driver: baby_driver)
+        Product.create!(name: 'Blood', uri: 'https://blood.com', manufacturer: orange_manufacturer, driver: taxi_driver)
+        Product.create!(name: 'Conference', uri: 'https://conference.com', manufacturer: pear_manufacturer, driver: baby_driver)
+        Product.create!(name: 'Concorde', uri: 'https://concorde.com', manufacturer: pear_manufacturer, driver: taxi_driver)
+      end
+
+      shared_context 'resource current_database' do
+        before do
+          connection = resource.connection
+
+          def connection.current_database
+            'db/test.sqlite3'
+          end
+        end
+
+        after do
+          resource.connection.singleton_class.remove_method(:current_database)
+        end
```

Comment:
I would like to edit this to represent a little bit more what's going on behind the scene. 

```suggestion
      shared_context 'resource current_database' do
        before do
          resource_connection = resource.connection
          target_connection = resource.reflect_on_association(:driver).klass.connection

          def resource_connection.current_database
            'db/test.sqlite3'
          end

          def target_connection.current_database
            'db/test3.sqlite3'
          end
        end

        after do
          resource.connection.singleton_class.remove_method(:current_database)
          resource.reflect_on_association(:driver).klass.connection.singleton_class.remove_method(:current_database)
        end
      end
```